tools/vscode-azurewebpubsub/src/workflows/hubSetting/common/CreateOrUpdateEventHandlerStep.ts

```diff
@@ -0,0 +1,96 @@
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { type IAzureUserInput } from "@microsoft/vscode-azext-utils";
+import { AzureWizardPromptStep } from "@microsoft/vscode-azext-utils";
+import { isUrlValid, localize } from "../../../utils";
+import { type ICreateOrUpdateHubSettingContext } from "../../common/contexts";
+import { KnownUserEvents, eventHandlerSystemEvents, eventHandlerUserEvents } from "../../../constants";
+
+export class CreateOrUpdateEventHandlerStep extends AzureWizardPromptStep<ICreateOrUpdateHubSettingContext> {
+    public constructor(private readonly isNewHandler: boolean) { super(); }
+
+    public async prompt(context: ICreateOrUpdateHubSettingContext): Promise<void> {
+        if (!(context.hubProperties?.eventHandlers)) {
+            throw new Error(localize(
+                'inValidICreateOrUpdateHubSettingContext',
+                'Invalid ICreateOrUpdateHubSettingContext, hubProperties = {0}',
+                JSON.stringify(context.hubProperties)
+            ));
+        }
+        let updateIndex = -1;
+        if (!this.isNewHandler) {
+            if (!(context.hubProperties?.eventHandlers?.length)) {
+                throw new Error(localize('emptyEventHandler', 'The event handler list is empty.'));
+            }
+            const choices = context.hubProperties.eventHandlers.map((handler, index) => ({ label: handler.urlTemplate, data: index}));
+            updateIndex = (await context.ui.showQuickPick(choices, { placeHolder: localize('selectEventHandler', 'Select Event Handler') })).data;
+            let eventHandler = context.hubProperties.eventHandlers[updateIndex];
+            eventHandler = {
+                ...eventHandler,
+                urlTemplate: await inputUrlTemplate(context.ui, eventHandler.urlTemplate),
+                userEventPattern: await inputUserEvents(context.ui),
+                systemEvents: await selectSystemEvents(context.ui)
+            };
+            context.hubProperties.eventHandlers[updateIndex] = eventHandler;
+        }
+        else {
+            context.hubProperties.eventHandlers.push({
+                urlTemplate: await inputUrlTemplate(context.ui),
+                userEventPattern: await inputUserEvents(context.ui), 
+                systemEvents: await selectSystemEvents(context.ui), 
+                auth: {}
+            });
+        }
+    }
+
+    public shouldPrompt(_context: ICreateOrUpdateHubSettingContext): boolean { return true; }
+}
+
+async function inputUrlTemplate(ui: IAzureUserInput, defaultUrlTemplate: string = "https://") {
+    return await (ui.showInputBox({
+        prompt: localize('enterEventHandlerUrlTemplate', 'Enter an URL Template for the event Handler.'),
```

Comment:
a URL, since U has a consonant sound :)
```suggestion
        prompt: localize('enterEventHandlerUrlTemplate', 'Enter a URL Template for the event Handler.'),
```